data/src/test/java/org/apache/iceberg/data/orc/TestGenericData.java

```diff
@@ -71,6 +72,30 @@ protected void writeAndValidate(Schema schema) throws IOException {
     for (int i = 0; i < expected.size(); i += 1) {
       DataTestHelpers.assertEquals(schema.asStruct(), expected.get(i), rows.get(i));
     }
+
+    // Also write and validate where all rows are the same (repeating)
+    List<Record> expectedRepeating = Collections.nCopies(100, expected.get(0));
```

Comment:
I think this should be a separate test case.

Reply:
Fixed